src/BucketTesting/Domain/Model/BucketLog.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\BucketTesting\Domain\Model;
+
+use DateTime;
+use DateTimeInterface;
+
+class BucketLog {
```

Comment:
Let's talk about naming (of this class) and namespace hierarchies. I like that you introduced the `Domain\Model` namespace, in one of the last commits, you should also move `Campaign`, `Bucket` and `CampaignDate` there. But if we do that, then our future selves will ask "What's the difference between a `Bucket` and a `BucketLog` and how are they related to bucket events?" 

I'd be ok with leaving class name and the logging namespace like this, but maybe we can find a better hierarchy.